sdk/src/Handlers/AwsSdk/Internal/XRayPipelineHandler.cs

```diff
@@ -104,17 +94,65 @@ public XRayPipelineHandler(Stream stream)
                 throw new ArgumentNullException("recorder");
             }
 
-            if (stream == null)
+            AWSServiceHandlerManifest = GetAWSServiceManifest(stream);
+        }
+
+        /// <summary>
+        /// Creates instance of <see cref="XRayPipelineHandler"/> with provided AWS service manifest instance.
+        /// </summary>
+        /// <param name="awsServiceManifest">Instance of <see cref="AWSServiceHandlerManifest"/></param>
+        public XRayPipelineHandler(AWSServiceHandlerManifest awsServiceManifest)
+        {
+            _recorder = AWSXRayRecorder.Instance;
+
+            if (_recorder == null)
             {
-                _logger.DebugFormat("The provided stream is null, initializing with default AWS whitelist.");
-                InitWithDefaultAWSWhitelist(_recorder);
+                throw new ArgumentNullException("recorder");
```

Comment:
It looks like if the recorder is missing when the `XRayPipelineHandler` is instantiated we'll throw an exception that isn't caught. Is this the same behavior as before the custom manifest was supported, or should it be logged instead?

Reply:
This is a new constructor and inline with existing [constructor](https://github.com/aws/aws-xray-sdk-dotnet/blob/c9189893d31d692d4f25d6adb4b0f2143d41dd5e/sdk/src/Handlers/AwsSdk/Internal/XRayPipelineHandler.cs#L70) which maintains the same behavior.